diesel/src/mysql/connection/bind.rs

```diff
@@ -0,0 +1,74 @@
+extern crate mysqlclient_sys as ffi;
+
+use mysql::MysqlType;
+use std::mem;
+use std::os::{raw as libc};
+
+pub struct Binds {
+    data: Vec<Option<Vec<u8>>>,
+    lengths: Vec<libc::c_ulong>,
+    is_nulls: Vec<ffi::my_bool>,
+    mysql_binds: Vec<ffi::MYSQL_BIND>,
+}
+
+impl Binds {
+    pub fn from_input_data(input: Vec<(MysqlType, Option<Vec<u8>>)>) -> Self {
+        let lengths = input.iter().map(|x| match *x {
+            (_, Some(ref data)) => data.len() as libc::c_ulong,
+            (_, None) => 0,
+        }).collect();
+        let is_nulls = input.iter()
+            .map(|&(_, ref data)| if data.is_none() { 1 } else { 0 })
+            .collect();
+        let (types, data) = input.into_iter().unzip::<_, _, Vec<_>, _>();
```

Comment:
Might be easier to read like `let (types, data): (Vec<MysqlType>, Vec<Option<Vec<u8>>>) = …` instead of turbofish